### ISSUE TITLE
early stopping이 채점하는 fold를 eval_set으로 쓴다 — CV 낙관 편향 + 제출 경로 불일치

### ISSUE THREAD

**Issue by ikazen** [ikazen/reflexion-rondo#305]
## 증상

CV가 개선되는 방향이 LB로 전달되지 않는다. 특히 `n_estimators`가 큰 후보와 LightGBM/CatBoost 후보가 CV에서 체계적으로 유리하게 평가되는데, 그 이득이 제출에서 사라지거나 역전된다.

## 원인

`evaluator/harness.py:186-211`

```python
def _fit_with_early_stopping(model, Xtr, ytr, Xva, yva) -> None:
    sig_params = inspect.signature(model.fit).parameters
    if "eval_set" in sig_params:
        kwargs: dict = {"eval_set": [(Xva, yva)]}
        if "callbacks" in sig_params:
            import lightgbm as lgb
            kwargs["callbacks"] = [lgb.early_stopping(_EARLY_STOPPING_ROUNDS, verbose=False)]
        elif "early_stopping_rounds" in sig_params:
            kwargs["early_stopping_rounds"] = _EARLY_STOPPING_ROUNDS
        model.fit(Xtr, ytr, **kwargs)
```

`evaluate_pipeline`의 fold 루프가 넘기는 `Xva/yva`는 **그 fold의 점수를 매기는 바로 그 데이터**다(`harness.py:894-905`). 트리 개수를 검증셋에서 고르고 같은 검증셋에서 채점하므로 `fold_scores`에 낙관 편향이 들어간다.

제출/holdout 경로는 정반대다. `harness.py:226`:

```python
if yva is not None:
    _fit_with_early_stopping(model, Xtr, ytr, Xva, yva)   # CV
    return model
model.fit(Xtr, ytr)                                       # 제출/holdout — 조기종료 없음
```

`bin/submit.py:_bagged_predict`가 `fit_predict(..., yva=None)`으로 부르고, `runtime/runner.py:_eval_holdout`도 마찬가지다. 즉 CV에서 300그루로 조기종료된 모델이 제출에서는 `n_estimators` 전량(예: 1000그루)으로 학습된다.

### 파급 세 가지

1. **후보 순위 오염.** 편향의 크기가 후보마다 다르다(낮은 learning_rate + 큰 n_estimators일수록 크다). ADR-047 실측 기준 s6e8의 p50→p90 경쟁 격차가 0.00553 AUC인데, 조기종료 낙관은 그 수준과 같은 자릿수다.
2. **모델군 비대칭.** `harness.py:193-195` 주석이 이미 인정하듯 XGBoost 3.x는 `fit()`에 `callbacks`/`early_stopping_rounds`가 없어 조기종료가 안 걸린다. LightGBM(`callbacks`)·CatBoost(`early_stopping_rounds`)만 걸린다. → `model_swap`이 일반화 성능과 무관한 이유로 LGBM/CatBoost를 선호한다.
3. **Optuna 레인 왜곡.** `evaluator/tuner.py`도 같은 `evaluate_pipeline`을 쓴다. `evaluator/search_spaces.py:19,35,48`이 `n_estimators`/`iterations`를 100~1000으로 탐색하는데, CV에선 조기종료가 이 값을 무력화하고 제출에선 그대로 쓴다 → 큰 값을 체계적으로 선호한다.
4. **cross-seed 게이트가 이걸 못 잡는다.** 편향은 모든 seed에서 동일하게 재현되므로 `_cross_seed_confirm`을 100% 통과한다. 유일하게 볼 수 있는 신호가 holdout인데 그건 이진 veto다(#별도 이슈).
5. **stacking 이중 불일치.** `_fit_predict_stack`은 meta를 조기종료된 멤버의 OOF로 학습시키고(`_oof_member_predictions`는 항상 inner-valid를 yva로 넘김), 제출 시엔 조기종료 없는 멤버의 예측을 입력받는다.

## 수정

가장 단순하고 안전한 방향은 **채점 fold를 eval_set에서 떼는 것**. 두 안 중 택일:

- (A) CV fold fit에서 조기종료를 아예 쓰지 않는다. `_fit_with_retry`의 `yva is not None` 분기를 제거하고 세 경로(CV/holdout/제출)를 `model.fit(Xtr, ytr)` 하나로 통일. 가장 적은 코드, 즉시 3경로 일치.
- (B) train fold를 다시 쪼개 inner-valid로 조기종료하고, 확정된 `best_iteration_`을 `selected_params`에 실어 제출 경로가 같은 트리 수로 학습하게 한다. 정확하지만 fold당 fit 비용과 배선이 늘어난다.

(A)를 기본으로 하고, 조기종료가 실제로 필요하면 (B)를 후속으로 본다.

## 영향 범위

`raw.pipelines.cv_score`가 전부 다른 스케일이 되므로 확정 baseline 재측정 필요(`bin/establish_baseline.py --remeasure`). ADR-040의 `train_fingerprint` 재확립과 동일 절차. `_EARLY_STOPPING_KEYS` 재시도 로직(#71/#239)은 (A)에서도 제출 경로에 그대로 필요하다.

parent: #304
